src/Execution/lbsTemperedWorkCriterion.py

```diff
@@ -49,28 +49,36 @@
 from src.Model.lbsRank import Rank
 
 
-class TemperedLoadCriterion(CriterionBase):
+class TemperedWorkCriterion(CriterionBase):
     """A concrete class for the Grapevine criterion modified in line 6
     """
 
-    def __init__(self, ranks, _, parameters):
-        """Class constructor:
-        ranks: set of ranks (lbsRank.Rank instances)
-        parameters: parameters dictionary needed for this criterion
+    def __init__(self, work_model, parameters: dict):
+        """Class constructor
+        parameters: optional parameters dictionary
         """
 
         # Call superclass init
-        super(TemperedLoadCriterion, self).__init__(ranks, _)
-        print(f"{bcolors.HEADER}[TemperedLoadCriterion]{bcolors.END} Instantiated concrete criterion")
+        super(TemperedWorkCriterion, self).__init__(work_model, parameters)
+        print(f"{bcolors.HEADER}[TemperedWorkCriterion]{bcolors.END} Instantiated concrete criterion")
 
-        # Use either actual or locally known destination loads
-        self.actual_dst_load = parameters.get("actual_destination_load", False)
+        
+        # Determine how destination work is to be computed
+        def get_dst_work_know_by_src(p_src, p_dst):
+            return p_src.get_known_work(p_dst)
+        def get_actual_dst_work(_, p_dst):
+            return self.get_work(p_dst)
+
+        # Retrieve releavant parameter when available
+        self.dst_work = get_dst_work_know_by_src if not (
```

Comment:
Don't you have to pass arguments to the functions `get_dst_work_know_by_src` and `get_actual_dst_work`

Reply:
Or you just want to `self.dst_work` to be a function?

Reply:
Yes, it has to be a function, used in the `compute()` method of the concrete criterion:
```
    def compute(self, obj: Object, p_src: Rank, p_dst: Rank) -> float:
        """Tempered work criterion based on L1 norm of works
        """

        # Criterion only uses object and rank works
        return self.get_work(p_src) - (
            self.dst_work(p_src, p_dst) + obj.get_time())
```
For efficiency's sake, we do not want this `if..else..` made at each invocation of `compute()` (as was the case before). Instead, we call the function that was properly initialized only once at construction time.